src/Stratis.Bitcoin.Features.BlockStore/BlockStoreFeature.cs

```diff
@@ -130,9 +126,7 @@ public override async Task InitializeAsync()
             if (!this.network.Consensus.IsProofOfStake)
                 this.connectionManager.Parameters.Services |= NetworkPeerServices.NODE_WITNESS;
 
-            this.signals.SubscribeForBlocksConnected(this.blockStoreSignaled);
-
-            return;
+            this.blockStoreSignaled.Initialize();
```

Comment:
it's a bit annoying that you have to call initialize on this object, even though you just received it in the ctor. I would assume that it's already ready for use.

Reply:
You can avoid that by subscribing right away bet I've kept the original logic, otherwise it wouldn't be a refactoring